src/content/docs/en/guides/testing.mdx

```diff
@@ -16,6 +16,106 @@ A Vite-native unit test framework with ESM, TypeScript and JSX support powered b
 
 See the [Astro + Vitest starter template](https://github.com/withastro/astro/tree/latest/examples/with-vitest) on GitHub.
 
+## Cypress
+
+Cypress is a front-end testing tool built for the modern web. Cypress enables you to write end-to-end tests for your Astro site.
+
+### Installation
+
+You can install Cypress using the package manager of your choice.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install -D cypress
+  ```
+
+  This will install Cypress locally as a dev dependency for your project.
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm add cypress --save-dev
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add cypress --dev
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Configuration
+
+In the root of your project, create a `cypress.config.js` file with the following content:
+
+```js title="cypress.config.js"
+import { defineConfig } from 'cypress'
+
+export default defineConfig({
+  e2e: {
+    supportFile: false
+  }
+})
+```
+
+### Create your first Cypress test
+
+1. Choose a page to test. We will use the example `index.astro` page below.
+
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
+      <head>
+        <title>Astro is awesome!</title>
+        <meta name="description" content="Pull content from anywhere and serve it fast with Astro's next-gen island architecture." />
+      </head>
+      <body>
+      <h1>Hello world from Astro</h1>
+      </body>
+    </html>
+    ```
+
+2. Create an `index.spec.js` file in the `cypress/e2e` folder. Use the following test into the file to verify that the page title and header are correct.
+
+    ```js title="cypress/e2e/index.cy.ts"
+    it('titles are correct', () => {
+      const page = cy.visit('http://localhost:3000');
+
+      page.get('title').should('have.text', 'Astro is awesome!')
+      page.get('h1').should('have.text', 'Hello world from Astro');
+    });
+    ```
+
+    :::tip[set a baseUrl]
+    You can set [`"baseUrl": "http://localhost:3000"`](https://docs.cypress.io/guides/end-to-end-testing/testing-your-app#Step-3-Configure-Cypress) in the `cypress.config.ts` configuration file to use `cy.visit("/")` instead of `page.goto("http://localhost:3000/")` for a more convenient URL.
+    :::
+
+### Running your Cypress tests
+
+Cypress can be run from the command line or from the Cypress App. The App provides a visual interface for running and debugging your tests.
+
+First, start the dev server so Cypress can access your live site. To run our test from the previous example using the command line, execute the following command:
+
+```shell
+npx cypress run
+```
+
+To run the test using the Cypress App, execute the following command:
+
+```shell
+npx cypress open
+```
+
+Once the Cypress App is launched, choose **E2E Testing**, then select the browser to be used to run tests.
+
+### Next steps
```

Comment:
Before next steps, it might be nice to briefly describe what kind of test results/feedback (and where) one should expect to receive.

I'll again put this as non-blocking, but the point of testing is to be able to respond to passing/failing tests! So, I think it would be important for this guide to let someone know what should happen if they've done this correctly, how to see their test results etc. In the example above, what happens when the test passes? Let's tell the reader so they know they did everything correctly!